features/storage/TESTS/kvstore/general_tests_phase_2/main.cpp

```diff
@@ -133,8 +133,8 @@ static void kvstore_init()
         erase_size = sec_bd->get_erase_size();
         // We must be able to hold at least 10 small keys (20 program sectors) and master record + internal data
         // but minimum of 2 erase sectors, so that the garbage collection way work
-        ul_bd_size  = align_up(program_size * PAGES_ESTIMATE, erase_size * 2);
-        rbp_bd_size = align_up(program_size * PAGES_ESTIMATE, erase_size * 2);
+        ul_bd_size  = align_up(program_size * PAGES_ESTIMATE, erase_size * 4);
```

Comment:
Should we actually set some kind of upper limit for the size, maybe 32KiB?

Reply:
Why `PAGES_ESTIMATE` is not a good limit?

Reply:
It was 8kB I had to increase to 16kB because     set_add_data_set_key_value_five_Kbytes  fails.
It seems that now is enough.

Reply:
PAGES_ESTIMATE is 40

 program_size  = sec_bd->get_program_size();
For QSPI  this return MBED_CONF_QSPIF_QSPI_MIN_PROG_SIZE  is 4  

Maybe we should modify this ?

Reply:
erase_size is 4kB

Reply:
I guess @kyle-cypress would disagree about dropping the estimate, at least some of the Cypress boards have really small(512B) sector size

Reply:
Indeed, many of the Cypress boards with internal flash only have sector size = page size = 512K.
Another comment: 
Why does this change the size to be aligned up to 4 * erase_size instead of 2. The intent of the current code is that PAGES_ESTIMATE reflects the main space requirement for the test. The align up to erase_size * 2 is just there because garbage collection requires two sectors in order to function properly, and on devices with a high sector to page size ratio PAGES_ESTIMATE may fit entirely in one sector.
If PAGES_ESTIMATE proves to be too low, I think the correct thing to do is to to increase PAGES_ESTIMATE. This will enable the size of the pre-test erase to adjust correspondingly.

Reply:
Reverted alignment. Fixed page size for  STM and Nordic boards Qspi flash page size